src/ts/types.ts

```diff
@@ -1,8 +1,10 @@
+import { ReactType } from 'react';
+
 export interface ComponentProps {
   /**
    * Set the component to render a different element type.
    */
-  component?: string;
+  component?: ReactType;
```

Comment:
This type is deprecated (and I don't think correct). You want an explicit list of available elements really... I can't remember where I've got this from before...

You may be able to use `keyof IntrinsicElements` or `Pick<keyof IntrinsicElements , 'div' | 'span' | 'etc etc'>`.

Or just manually define them: `div | span | etc etc`